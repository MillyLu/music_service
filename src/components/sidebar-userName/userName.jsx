import { useContext, useState, useEffect } from "react";
import { ThemeContext } from "../../ThemeProvider";
import * as Styled from './styles'

export function UserName() {
    const {theme} = useContext(ThemeContext);
    const [name, setName] = useState('');
    useEffect (() => {
        if(localStorage.getItem('name')) {
        setName(localStorage.getItem('name'))
    }
},[]);

    return(
        <Styled.SidebarPersonalName theme={theme}>{name}</Styled.SidebarPersonalName>
    )
}