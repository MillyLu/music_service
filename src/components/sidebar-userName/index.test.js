/* eslint-disable no-undef */
import * as ShallowRenderer from 'react-shallow-renderer';
// import { ShallowRenderer, createRenderer } from 'react-shallow-renderer';
import { UserName } from './userName';



describe('<UserName />', () => {
        it('should render successfully and match snapshot', () => {
            const shallowRenderer = ShallowRenderer.createRenderer();
            const renderer = shallowRenderer;
            renderer.render(<UserName  />);
            const result = renderer.getRenderOutput();
            expect(result).toMatchSnapshot();
          })
      });
