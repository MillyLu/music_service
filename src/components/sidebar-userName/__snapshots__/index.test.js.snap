// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserName /> should render successfully and match snapshot 1`] = `
<styled.p
  theme={0}
>
  
</styled.p>
`;
